pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -300,18 +300,21 @@ func (f *Fit) isSchedulableAfterPodChange(logger klog.Logger, pod *v1.Pod, oldOb
 	return framework.Queue, nil
 }
 
-// isResourceScaleDown checks whether the resource request of the modified pod is less than the original pod
-// for the resources requested by the pod we are trying to schedule.
-func (f *Fit) isResourceScaleDown(targetPod, originalOtherPod, modifiedOtherPod *v1.Pod) bool {
-	if modifiedOtherPod.Spec.NodeName == "" {
-		// no resource is freed up whatever the pod is modified.
+// isResourceScaleDown checks whether an update event may make the pod schedulable. Specifically:
+// - Returns true when an update event shows a scheduled pod's resource request got reduced.
+// - Returns true when an update event is for the unscheduled pod itself, and it shows the pod's resource request got reduced.
+func (f *Fit) isResourceScaleDown(targetPod, originalPod, modifiedPod *v1.Pod) bool {
+	if !(modifiedPod.Namespace == targetPod.Namespace && modifiedPod.Name == targetPod.Name) &&
```

Comment:
I don't think this check is actually necessary. Unscheduled Pods don't generally trigger an event. WDYT @sanposhiho?
Only when they are waiting https://github.com/kubernetes/kubernetes/blob/adc256ff0465130ecc6fb8336d827e7d54a42bed/pkg/scheduler/eventhandlers.go#L186

Reply:
For now, that's correct. We don't trigger any Pod related events basically if Pods are unscheduled. 
But, ideally, we should inform QHints of all Pod-related events regardless of whether they're scheduled or not. 
It's a wrong assumption that unscheduled Pods-related event shouldn't make any Pods schedulable. For in-tree plugins, that's correct probably, but out-of-tree plugins QHint may want to subscribe such events. (Actually, my company's custom plugins do, see [this](https://github.com/kubernetes/kubernetes/issues/110175#issuecomment-1140397251)) 

So, I'm inclined to move towards that way in the future, and I'd like to keep this `if` as it is.